### PR TITLE
Agreement: Simplify dispute metadata construction

### PR DESCRIPTION
**Pull request by bingen**
Besides using abi.encodePacked, there’s no need for a semi-colon as
fields are fixed length.

### CODE REVIEW

File: apps/agreement/contracts/Agreement.sol [hunk 1]
```diff
@@ -383,7 +380,7 @@ contract Agreement is IAgreement, AragonApp {
         require(msg.sender == submitter, ERROR_SENDER_NOT_ALLOWED);
 
         IArbitrator arbitrator = _getArbitratorFor(action);
-        bytes memory metadata = _buildDisputeMetadata(action);
+        bytes memory metadata = abi.encodePacked(appId(), action.currentChallengeId);
```
Comment (sohkai) [inline apps/agreement/contracts/Agreement.sol:383]: I do still like having a `:`, if only to make things a bit more obvious when looking at the raw bytes.

Would we be able to do this with `abi.encodePacked(appId(), ":", action.currentChallengeId)`?

Comment (sohkai) [reply apps/agreement/contracts/Agreement.sol:383]: Oh nevermind, I see https://github.com/aragon/aragon-apps/pull/1189!

How much worse in gas is it to do avoid the `:`?

Comment (facuspagnuolo) [reply apps/agreement/contracts/Agreement.sol:383]: I'll re-open the other one, I was inclined due to the gas costs, but actually I like the other one better. Let's check the gas costs diff and decide!

Comment (bingen) [reply apps/agreement/contracts/Agreement.sol:383]: Yes, that’s #1189

Comment (bingen) [reply apps/agreement/contracts/Agreement.sol:383]: Do you always remember by heart that `3a` corresponds to the colon? :astonished:

**Pull request merged**